frontend/src/commit-statuses.tsx

```diff
@@ -1,38 +1,36 @@
 import { Pull } from './pull';
-import styled from 'styled-components';
+import { StatusState, CommitStatus } from './types';
+import { Box, Link, chakra, useStyleConfig} from "@chakra-ui/react"
+import styled from "@emotion/styled"
 
-const padding = 5;
-const width = 10;
+const statusSize = 10;
 const marginBetween = 4;
+const horizontalMargin = 4;
 
-export const Status = styled.a`
-   width: 100%;
-   height: 100%;
-   margin: ${marginBetween / 2}px 0;
-   border-radius: 5px;
-   background: green;
-   display: none;
-`;
+const Status = chakra.a;
 
 const StatusContainer = styled.div`
-   position: absolute;
-   left: ${padding}px;
-   top: ${padding}px;
-   bottom: ${padding}px;
-   width: ${width}px;
+   margin: 0 ${horizontalMargin}px 0 ${horizontalMargin}px;
+   width: ${statusSize}px;
    display: flex;
-   justify-content: flex-start;
+   flex-shrink: 0;
    flex-direction: column;
+   justify-content: space-between;
+   gap: ${marginBetween}px;
 `;
 
 export function CommitStatuses({pull}: {pull: Pull}) {
    return (
-   <StatusContainer>
-      {pull.status.commit_statuses.map((status) => 
-         <Status key={status.data.context} href={status.data.target_url}>
-         </Status>
-      )}
+   <StatusContainer className="build_status_container">
+      {pull.buildStatuses().map((status) => {
+         const styles = useStyleConfig('Status', {variant: status.data.state});
+         return (<Status __css={styles}
```

Comment:
Haven't used `useStyleConfig` myself, but just going off the [docs](https://chakra-ui.com/docs/theming/advanced#single-part-component) this seems right.

I'm not sure what the difference between the `__css` and the `sx` props are.

Also I think `Link` is effectively an `<a>` tag, so you could do this too (instead of `const Status = chakra.a`)

```suggestion
         const styles = useStyleConfig('Status', {variant: status.data.state});
         return (<Link  sx={styles}
```

Reply:
Turns out there's a bug in chakra (seeimingly). `__css` and `sx` take the same kind of inputs, but `sx` has higher priority and will overwrite values from `__css`. Buuuut...  `_css` seems to be ignored when using Chakra components directly.

![image](https://user-images.githubusercontent.com/26855/144686558-275b08eb-72c3-4981-8bd8-f4146639bade.png)

Reply:
huh interesting